File: Leetcode 3401-3500/3427/3427.java
// Leetcode 3427: Sum of Variable Length Subarrays
// https://leetcode.com/problems/sum-of-variable-length-subarrays/
// Solved on 11th of November, 2025
class Solution {
    /**
     * Calculates the sum of variable-length subarrays.
     * For each element `nums[i]`, it sums the subarray from `max(0, i - nums[i])` to `i`.
     * @param nums The input array of integers.
     * @return The total sum of these variable-length subarrays.
     */
    public int subarraySum(int[] nums) {
        int n = nums.length;
        long[] prefix = new long[n + 1];
        for (int i = 0; i < n; i++) {
            prefix[i + 1] = prefix[i] + nums[i];
        }
        long total = 0;
        for (int i = 0; i < n; i++) {
            int start = i - nums[i];
            if (start < 0) {
                start = 0;
            }
            total += prefix[i + 1] - prefix[start];
        }
        return (int) total;
    }
}

File: Leetcode 3401-3500/3427/3427.py
# Leetcode 3427: Sum of Variable Length Subarrays
# https://leetcode.com/problems/sum-of-variable-length-subarrays/
# Solved on 11th of November, 2025
class Solution:
    def subarraySum(self, nums: list[int]) -> int:
        """
        Calculates the sum of variable length subarrays based on the problem description.

        Args:
            nums: A list of integers.
        Returns:
            The total sum of the variable length subarrays.
        """
        n = len(nums)
        prefixSum = [0] * (n + 1)

        for i in range(n):
            prefixSum[i + 1] = prefixSum[i] + nums[i]

        totalSum = 0
        for i in range(n):
            start = max(0, (i - nums[i]))
            currentSubarraySum = prefixSum[i + 1] - prefixSum[start]
            totalSum += currentSubarraySum

        return totalSum